nipype/utils/filemanip.py

```diff
@@ -780,8 +780,8 @@ def emptydirs(path, noexist_ok=False):
     try:
         shutil.rmtree(path)
     except OSError as ex:
-        elcont = os.listdir(path)
-        if ex.errno == errno.ENOTEMPTY and not elcont:
+        elcont = [p for p in os.listdir(path) if not p.startswith(".nfs")]
+        if ex.errno in [errno.ENOTEMPTY, errno.EBUSY] and not elcont:
```

Comment:
Can we try putting in a test to make sure this works as expected? Something like:

```Python
import os
from pathlib import Path
from unittest import mock

def nfs_unlink(pathlike, *, dir_fd=None):
    if dir_fd is None:
        path = Path(pathlike)
        deleted = path.with_name(".nfs00000000")
        path.rename(deleted)
    else:
        os.rename(pathlike, ".nfs1111111111", src_dir_fd=dir_fd, dst_dir_fd=dir_fd)

def test_emptydirs_dangling_nfs(tmp_path):
    busyfile = tmp_path / "base" / "subdir" / "busyfile"
    busyfile.parent.mkdir(parents=True)
    busyfile.touch()
    with mock.patch("os.unlink") as mocked:
        mocked.side_effect = nfs_unlink
        emptydirs(tmp_path / "base")

    assert Path.exists(tmp_path / "base")
    assert not busyfile.exists()
    assert busyfile.parent.exists()  # Couldn't remove
```

Reply:
That sounds like a great idea :-) Your code works as is, I just had to update my patch to also condsider the `subdir`. For the `nfs_unlink` mock, I also removed the case for `dir_fd` being `None`, because as far as I can tell, the `os.rename` line also handles that (`src_dir_fd` and `dst_dir_fd` are `None` by default).

Reply:
The reason for the other case is that, if we don't get the dir_fd, we would get more than just the basename.

Reply:
That makes sense! Thank you for explaining